src/Microsoft.ML.Recommender/SafeTrainingAndModelBuffer.cs

```diff
@@ -133,19 +133,19 @@ private struct MFParameter
             /// Specify if the factor matrices should be non-negative.
             /// </summary>
             [FieldOffset(48)]
-            public int DoNmf;
+            public sbyte DoNmf;
```

Comment:
Why `sbyte`?  Why not just `byte`?

Reply:
It's int8 in C side.

Reply:
Make it `uint8_t` on the C side then?

Signed-ness doesn't really matter here. So using `byte` is more appropriate.

Reply:
Ok. I will change other place to match this decision.